pdns/ueberbackend.cc

```diff
@@ -322,20 +349,43 @@ bool UeberBackend::getAuth(const DNSName &target, const QType& qtype, SOAData* s
   // of them has a more specific zone but don't bother asking this specific
   // backend again for b.c.example.com., c.example.com. and example.com.
   // If a backend has no match it may respond with an empty qname.
-
   bool found = false;
   int cstat;
   DNSName shorter(target);
   vector<pair<size_t, SOAData> > bestmatch (backends.size(), make_pair(target.wirelength()+1, SOAData()));
   do {
+    int zoneId{-1};
+    if(cachedOk && g_domainCache.isEnabled()) {
+      if (g_domainCache.getEntry(shorter, zoneId)) {
+        // Zone exists in domain cache, directly look up SOA.
+        // XXX: this code path and the cache lookup below should be merged; but that needs the code path below to also use ANY.
+        // Or it should just also use lookup().
+        DNSZoneRecord zr;
+        lookup(QType(QType::SOA), shorter, zoneId, nullptr);
+        if (!get(zr)) {
+          g_log << Logger::Info << "Backend returned no SOA for domain '" << shorter.toLogString() << "', which it reported as existing " << endl;
+          return false;
```

Comment:
continue; is the proper action here. Now you are hiding parent zones of disabled zone.
Please put the log message behind log-dns-details. Disabling a zone with a lot of qps will flood the logs and this might hurt performance for the other zones.